packages/machine/test/integration/test-io-provider.ts

```diff
@@ -34,70 +31,30 @@ export class TestIOProvider {
       serializedMessage
     ) as cf.legacy.node.ClientActionMessage;
 
-    let done = false;
-    const executedListeners = [] as number[];
-    let count = 0;
-
-    // Invoke all listeners waiting for a response to resolve their promise
-    this.listeners.forEach(listener => {
-      if (
-        listener.appId === message.appId ||
-        (!listener.appId && listener.multisig === message.multisigAddress)
-      ) {
-        listener.method(message);
-        done = true;
-        executedListeners.push(count);
-        count += 1;
-      }
-    });
-
-    // Now remove all listeners we just invoked
-    executedListeners.forEach(index => this.listeners.splice(index, 1));
-
-    // initiate ack side if needed
-    if (message.seq === 1) {
+    if (this.waitForIoContinuation !== undefined) {
+      const k = this.waitForIoContinuation;
+      this.waitForIoContinuation = undefined;
+      k(message);
+    } else if (message.seq === 1) {
```

Comment:
Can we document why this equality is being done?

It's kind of opaque without context.